google/cloud/forseti/scanner/audit/retention_rules_engine.py

```diff
@@ -353,15 +377,70 @@ def find_violations_in_bucket(self, bucket):
         """Get a generator for violations.
 
         Args:
-            bucket (bucket): Find violation from the buckets.
+            bucket (Bucket): Find violation from the buckets.
         Returns:
             Generator: All violations of the buckets breaking rules.
         """
-
         violation_max = self.bucket_max_retention_violation(bucket)
         violation_min = self.bucket_min_retention_violation(bucket)
         return itertools.chain(violation_max, violation_min)
 
+    def table_max_retention_violation(self, table):
+        """Get a generator for violations especially for maximum retention
+           It only supports bucket for now, and will work on generalizing
+           in future PRs.
+
+        Args:
+            table (Table): Find violation from the table.
+        Yields:
+            RuleViolation: All max violations of the table breaking the rule.
+        """
+        if self.max_retention is None:
+            return
+
+        table_dict = json.loads(table.data)
+        table_expiration = table_dict.get('expirationTime')
+        if not table_expiration:
+            self.generate_table_violation(table)
+
+        table_creation = table_dict.get('creationTime')
```

Comment:
if you're assuming creationTime is set then use table_dict['creationTime']. That way if this ever fails this line will make it clear which key is missing